### DISCUSSION TITLE
How can I make this keybinding with multiple assignments work?

### DISCUSSION THREAD

**Discussion by idr4n** [wezterm/wezterm#2397] [Q&A] [answered]
Hi,
I'm trying to open a pane to the left of the current one, and get the cursor back to the initial pane with the following keybinding:

```lua
{
	key = "<",
	mods = "CTRL|SHIFT",
	action = act.Multiple({
		act.SplitPane({ direction = "Left", size = { Percent = 30 } }),
		act.ActivatePaneDirection("Right"),
	}),
}
```

It does open the pane to the left, but leaves the cursor in the new pane. Any ideas why this is not working?

Thanks in advance!

**Comment by wez** [comment, marked as answer]
some actions (like SplitPane) are async, so when you group them in `Multiple`, the first one is scheduled to do something but hasn't fully started/completed by the time your second step is triggered.

I would suggest using `action_callback` to trigger a lua callback for this instead; take a look at: https://github.com/wez/wezterm/discussions/2388#discussioncomment-3371519

Using the lua mux api will let you create the split, then you can `window:perform_action` to activate in the direction you want.

This isn't perfect, but I think you can achieve something workable with it.

**Reply by idr4n** [reply]
Thanks a lot, I will give this a try!

**Reply by idr4n** [reply]
Yeap, this is working for me as you suggested:

```lua
	{
		key = "<",
		mods = "CTRL|SHIFT",
		action = wezterm.action_callback(function(win, pane)
			local newPane = pane:mux_pane()
			newPane:split({ direction = "Left", size = 0.3 })
			win:perform_action(act.ActivatePaneDirection("Right"), pane)
		end),
	},
```

Thanks again!

**Reply by idr4n** [reply]
One more thing. I was trying to get the new split pane to have the same working directory as the initial pane. I use the same keybinding as as above but changing the new split options to:

```lua
newPane:split({ direction = "Right", size = 0.3, cwd = pane:get_current_working_dir() })
```

This, however, opens the new split as desired but with the home directory as cwd, that is `pane:get_current_working_dir()` is not picking up the initial wd in this case.

**Reply by wez** [reply]
does spawning a new tab pick up the correct working directory? In other words, is it only splitting that doesn't pick it up?

**Reply by idr4n** [reply]
actually I just tried again the same binging with `newPane:split({ direction = "Left", size = 0.3 })` and it is picking up the current wd correctly, no need to add the `cwd` field (actually with the cwd field it does not pick it up correctly). New tabs pick up the wd correctly, so everything seems to be working just fine. Thanks a lot!

**Reply by idr4n** [reply]
OK, so I know why that was happening... If I am in a Tmux session, splitting or spawning tabs pick up the directory I was when I first started Tmux. Once in Tmux, neither splitting not new tabs pick up the current directory of the Tmux pane.
